Compilation error since @swc/core 1.3.79 on Nest.js projects

### Describe the bug

When updating from `@swc/core` from version `1.3.78` to `1.3.79` (and any version above), compilation fails with a brand new Nest.js project using `swc` as compiler and type-checker (configured inside `nest-cli.json` file).
The issue can be seen here: https://codesandbox.io/p/sandbox/dawn-hill-9858zs?file=/package.json:34,5-34,28 (but also encountered on my local machine)


**Steps to reproduce:**
- open a new terminal
- run `npm start`
- see that compilation fails with `Error: Cannot find module 'src/app.module'`
- modify `"@swc/core": "^1.3.87"` to `"@swc/core": "1.3.78"` (no caret) inside `package.json`
- run `npm i && npm start`
- see that it compiles successfully

### Input code

_No response_

### Config

_No response_

### Playground link

_No response_

### SWC Info output

Output from CodeSandBox terminal
```
    Operating System:
        Platform: linux
        Arch: x64
        Machine Type: x86_64
        Version: #1 SMP PREEMPT_DYNAMIC Sun Aug  6 20:05:33 UTC 2023
        CPU: (2 cores)
            Models: AMD EPYC

    Binaries:
        Node: 20.5.1
        npm: 9.8.0
        Yarn: 1.22.19
        pnpm: N/A

    Relevant Packages:
        @swc/core: 1.3.87
        @swc/helpers: N/A
        @swc/types: 0.1.5
        typescript: 5.2.2

    SWC Config:
        output: N/A
        .swcrc path: N/A

    Next.js info:
        output: N/A
```

### Expected behavior

Compilation should run successfully

### Actual behavior

Currently, the compilation fails with the following error:
```
Error: Cannot find module 'src/app.module'
Require stack:
- /workspace/dist/main.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:1048:15)
    at Function.Module._load (node:internal/modules/cjs/loader:901:27)
    at Module.require (node:internal/modules/cjs/loader:1115:19)
    at require (node:internal/modules/helpers:130:18)
    at Object.<anonymous> (/workspace/dist/main.js:6:20)
    at Module._compile (node:internal/modules/cjs/loader:1233:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1287:10)
    at Module.load (node:internal/modules/cjs/loader:1091:32)
    at Function.Module._load (node:internal/modules/cjs/loader:938:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:83:12)
```

### Version

Since @swc/core@1.3.79

### Additional context

_No response_

--- a/crates/swc/src/config/mod.rs
+++ b/crates/swc/src/config/mod.rs
@@ -1587,13 +1587,14 @@ impl ModuleConfig {
         config: Option<ModuleConfig>,
         available_features: FeatureFlag,
     ) -> Box<dyn swc_ecma_visit::Fold + 'cmt> {
+        let skip_resolver = base_url.as_os_str().is_empty() && paths.is_empty();
+
         let base = match base {
-            FileName::Real(v) if !paths.is_empty() => {
+            FileName::Real(v) if !skip_resolver => {
                 FileName::Real(v.canonicalize().unwrap_or_else(|_| v.to_path_buf()))
             }
             _ => base.clone(),
         };
-        let skip_resolver = base_url.as_os_str().is_empty() && paths.is_empty();
 
         match config {
             None => {

